inventory-upgrade.py

```diff
@@ -0,0 +1,350 @@
+                for pattern in DEPRECATED_VARS[version_depreciation]:
+                    for key in list(self.content.keys()):
+                        if re.match(pattern, key):
+                            del self.content[key]
+                            print(f"Warning: deprecated variable '{key}', removed.")
+
+        # Upgrade docker_registries syntax to simplify mandatory vars.yml content when possible
+        if (
+            "docker_registries" in self.content
+            and "docker_private_registry_login" not in self.content
+            and len(self.content["docker_registries"]) == 1
+        ):
+            current_registry = self.content["docker_registries"][0]
+            ref_registry = ref_values["docker_registries"][0]
+            if (
+                current_registry["url"] == ref_registry["url"]
+                and current_registry["password"] == ref_registry["password"]
+            ):
+                self.content["docker_private_registry_login"] = current_registry[
+                    "username"
+                ]
+                del self.content["docker_registries"]
+
+    def use_custom_tag(self):
+        """Returns True if the inventory contains custom images / tags"""
+        # TODO: this is not working when there are deprecated images in the inventory
+        # Not sure how to managed this case yet
+        ignored_images = [
+            "docker_rabbitmq.*",
+            "docker_postgres.*",
+            "docker_nginx.*",
+            "docker_acme_companion.*",
+            "docker_bimdata_tag",
+        ]
+        for key in self.content:
+            if re.match("docker_.*_tag", key):
+                if self.content[key] == "{{ docker_bimdata_tag }}":
+                    return True
+        return False
+
+    def backup(self, backup_suffix):
+        """Backup the inventory"""
+        backup_path = self.path.with_suffix(backup_suffix)
+        if backup_path.exists():
+            suffix = backup_suffix + "-" + str(int(round(datetime.now().timestamp())))
+            backup_path = self.path.with_suffix(suffix)
+            print(
+                f"Warning: {self.path.with_suffix(backup_suffix)} already exists.\n"
+                f"The inventory will be backup up into: {backup_path}\n"
+            )
+        shutil.copytree(self.path, backup_path)
+
+    def write_inventory(self, backup=True, backup_suffix=".backup", delete_legacy=True):
+        if self.is_legacy:
+            raise ValueError(
+                "This script can't write legacy inventory. Please upgrade it first."
+            )
+        if backup:
+            self.backup(backup_suffix)
+
+        # Segregate the mandatory variables from the custom ones, there are is the template.
+        # The custom variables are written at the end of the file
+        required_variables = {}
+        grouped_variables = []
+        previous_prefix = None
+        for key, value in self.content.items():
```

Comment:
If you can find a more precise name than `content`, it will help to read the script

Reply:
I've removed some other content, but this one is the content of the inventory on which we are working. It's read from the files, then eventually transformed, and finally written to the file. It seems appropriate in a class named Inventory. But I'm open to suggestions!